test/ERC20CosmosCoinAmountConversionTest.t.sol

```diff
@@ -0,0 +1,436 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.25 <0.9.0;
+
+// solhint-disable custom-errors,max-line-length
+import { Test } from "forge-std/Test.sol";
+import { ERC20CosmosCoinAmountConversion } from "../src/utils/ERC20CosmosCoinAmountConversion.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+// Discuss - Do we want to move mock contract to a mock folder?
+// Mock ERC20 token without the decimals function overridden
+contract MockERC20 is ERC20 {
+    constructor() ERC20("BaseToken", "BTK") {
+        _mint(msg.sender, 100_000_000_000_000_000_000); // Mint some tokens for testing
+    }
+}
+
+// Discuss - Do we want to move mock contract to a mock folder?
+// Mock ERC20 token with the decimals function overridden
+contract MockERC20Metadata is ERC20 {
+    uint8 private _decimals;
+
+    constructor(uint8 decimals_) ERC20("MetadataToken", "MTK") {
+        _decimals = decimals_;
+        _mint(msg.sender, 100_000_000_000_000_000_000); // Mint some tokens for testing
+    }
+
+    function decimals() public view override returns (uint8) {
+        return _decimals;
+    }
+}
+
+contract ERC20CosmosCoinAmountConversionTest is Test {
```

Comment:
Might be worth using both fuzz and invariants testing here: https://book.getfoundry.sh/forge/advanced-testing

Reply:
It's absolutely a great Idea! Only wonder, do we want to address this now? or later on? If you guys needs this lib asap to complete you workflows, I can address this in a later on PR and refactor the testing file. My primary goal was to ensure it works properly :)

Reply:
Let's address this now. We don't require this lib asap since our mock ERC20's size doesn't cause issues. So it is ok to take time to make sure we design and test this properly.